### PR TITLE
Adding notes on KafkaSource usage

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

## Proposed Changes 

- adding some notes/background info on usecase and potential limitations of the `KafkaSource`

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]
LGTM

**Review by slinkydeveloper — APPROVED** [review]

File: docs/eventing/samples/kafka/source/index.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ type: "docs"
 
 Tutorial on how to build and deploy a `KafkaSource` [Eventing source](../../../sources/README.md) using a Knative Serving `Service`.
 
+## Background
+
+The `KafkaSource` reads all the messages, from all partitions, and converts these Kafka Records to a CloudEvent and distributes those via HTTP to its configured `sink`.
```
Comment (lionelvillard) [inline docs/eventing/samples/kafka/source/index.md:12]: Do you want to say that when the record is already a CloudEvent, there is no conversion?
Do you want to say anything about QoS? at-least-once, partition ordering is preserved

Comment (matzew) [reply docs/eventing/samples/kafka/source/index.md]: updated the sentence a bit!

**Pull request merged**